executor/show.go

```diff
@@ -1233,8 +1233,23 @@ func appendPartitionInfo(partitionInfo *model.PartitionInfo, buf *bytes.Buffer,
 		return
 	}
 	if partitionInfo.Type == model.PartitionTypeHash {
-		fmt.Fprintf(buf, "\nPARTITION BY HASH( %s )", partitionInfo.Expr)
-		fmt.Fprintf(buf, "\nPARTITIONS %d", partitionInfo.Num)
+		fmt.Fprintf(buf, "\n/*!50100 PARTITION BY HASH (%s)", partitionInfo.Expr)
+		if partitionInfo.Num <= 1 {
```

Comment:
Will this not hide something like this:
```sql
PARTITION BY HASH (id)
(PARTITION pOnly COMMENT = 'a single partition')
```

I'm not sure how easy it is to be fully compatible with MySQL, since IIRC it will store the partitioning definition as text, making it easy to check if it is the default 'PARTITIONS N' or all partitioned defined one by one.

I also know it is a bit strange to define each hash partition, since it should be possible to do 'ADD PARTITION' or COALESCE PARTITION' which would simply add (a possible default partition) or drop one or more partitions and redistribute all the rows/data. But as long as we define how TiDB handles these I'm OK with which behaviour we choose.
I.e. differentiate between default 'PARTITIONS N' and define each and every partition
OR always show the definition of each and every partition in the `SHOW CREATE TABLE` output.

Reply:
yes, that makes me puzzled too, so I chose to show the only partition first, but for compatibility, I hid it then.
IMO, I prefer to list partitions one by one, including the only parition. that make partitions definition more clear.

Reply:
related to https://github.com/pingcap/tidb/pull/8477

Reply:
I created another PR to cover a bit more: https://github.com/pingcap/tidb/pull/29945
Do you think that PR would cover all this as well?

Reply:
LGTM, the nested annotation is a problem indeed